python_modules/libraries/dagster-k8s/dagster_k8s/container_context.py

```diff
@@ -204,6 +215,218 @@ def get_environment_dict(self) -> Mapping[str, str]:
         parsed_env_var_tuples = [parse_env_var(env_var) for env_var in self.env_vars]
         return {env_var_tuple[0]: env_var_tuple[1] for env_var_tuple in parsed_env_var_tuples}
 
+    def _snake_case_allowed_fields(
+        self, only_allow_user_defined_k8s_config_fields: Mapping[str, Any]
+    ) -> Mapping[str, Any]:
+        result = {}
+
+        for key in only_allow_user_defined_k8s_config_fields:
+            if key == "namespace":
```

Comment:
would the dagster agent have the permissions to report on the pod status if it's provisioned in a different namespace?

Reply:
The user would need to grant it those permissions (e.g. via a cluster role binding or I think its possible to individually grant permissions).

Reply:
yes, I guess I poorly formulated my question. I meant using the reference deployment / helm chart. I guess what I am getting at is, is there a valid use case for this, and would we need to do more elsewhere to support it or is it better to disallow this ? 

nit, not a blocker. just a thought

Reply:
We do have users who want to run different code locations in different namespaces - e.g. a single agent serving multiple teams, each with their own namespace and service account